src/main/java/de/retest/recheck/ui/descriptors/RootElementFilter.java

```diff
@@ -0,0 +1,96 @@
+package de.retest.recheck.ui.descriptors;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.ui.image.Screenshot;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class RootElementFilter {
+	private final Filter filter;
+
+	public RootElement filter( final RootElement rootElement ) {
+		if ( filter.equals( Filter.FILTER_NOTHING ) ) {
```

Comment:
Maybe extract a separate method `filterNothing()`?

Reply:
Removing this check would mean a new `RootElement` instance is created with `FILTER_NOTHING`. I believe extracting this logic to a separate method is a better idea